staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/cbor.go

```diff
@@ -218,6 +218,16 @@ func (s *serializer) unmarshal(data []byte, into interface{}) (strict, lax error
 			}
 		}()
 		into = &content
+	} else {
+		// Decoding to Unstructured never produces values with dynamic type RawExtension, so
+		// the transcode step is only necessary for non-Unstructureds.
+
+		// TODO: Option to disable this behavior.
+		defer func() {
```

Comment:
no.  Please use a bool and inline the code for transcoding below the into

```go
	if errors.As(err, &unknownField) {
		// Unlike JSON, there are no strict errors in CBOR for duplicate map keys. CBOR maps
		// with duplicate keys are considered invalid according to the spec and are rejected
		// entirely.
		return runtime.NewStrictDecodingError([]error{unknownField}), modes.DecodeLax.Unmarshal(data, into)
	}
	if err != nil{
		return nil, err
	}
	if needTranscodeRawTypes{
		if err := transcodeRawTypes(into); err != nil {
			return errors here
		}
	}
	return nil, nil
```

Reply:
Thanks for pointing this out. I agree this is misplaced, and I think it belongs in Decode after the call to unmarshal.